### ISSUE TITLE
Links in document outline point to wrong position if complex YAML header included

### ISSUE THREAD

**Issue by salim-b** [mangecoeur/document-outline#42]
I stumbled upon another issue related to the YAML metadata block. If a Markdown file contains a YAML header with some long, "complex" entries like the following, the links in the document outline take you to the wrong position in the file when clicking them.

```
---
header-includes:
  - <!-- remap unsupported unicode chars -->
  - \usepackage{newunicodechar}
  - \newunicodechar{✅}{✔}
  - \newunicodechar{❌}{✘}

  - <!-- define titlepage as cover, see KOMA Script manual p. 296 -->
  - \KOMAoptions{titlepage = firstiscover}
  
  - <!-- avoid period after table/figure (and chapter/section etc.) number, see KOMA Script manual p. 103f or <https://tex.stackexchange.com/questions/29181/figure-and-table-numbers-in-caption-are-terminated-by-a-period-and-semicolon> -->
  - \KOMAoptions{numbers = noendperiod}

  - <!-- customize chapter font (H1) -->
  - \addtokomafont{chapter}{\scshape}

  - <!-- change color of `inline code spans` -->
  - \usepackage[dvipsnames,table]{xcolor}
  - \definecolor{bckgrnd}{RGB}{248,248,248}
  - \let\OldTexttt\texttt
  - \renewcommand{\texttt}[1]{\OldTexttt{\color{Gray}{#1}}}

  - <!-- customize (syntax highlighted) code fences -->
  - \usepackage{fancyvrb}
  - \DefineVerbatimEnvironment{Highlighting}{Verbatim}{
      commandchars = \\\{\},
      fontsize = \scriptsize,
      fontseries = l,
      baselinestretch = 1
    }
  - \DefineVerbatimEnvironment{verbatim}{Verbatim}{
      commandchars = \\\{\},
      fontsize = \scriptsize,
      fontseries = l,
      formatcom = \color{Gray},
      baselinestretch = 1
    }

  - <!-- customize footnotes -->
  - \deffootnote[3em]{3em}{0em}{\makebox[2em][l]{\textsuperscript\thefootnotemark}}
  - \setlength{\footnotesep}{1.2em}
  - \setlength{\skip\footins}{1.8em}
  - \setlength{\footskip}{22pt}
  
  - <!-- customize header and footer -->
  - \pagestyle{headings}
  - \usepackage{scrlayer-scrpage}
  - \KOMAoptions{headsepline = on}
  - \KOMAoptions{footsepline = on}
  - \KOMAoptions{plainfootsepline = on}
  - \addtokomafont{pageheadfoot}{\normalfont\small}
  - \addtokomafont{pagenumber}{\small}
  - \ModifyLayer[addvoffset = 3pt]{scrheadings.head.below.line}
  - \ModifyLayer[addvoffset = 3pt]{plain.scrheadings.head.below.line}
  
  - <!-- customize table and figure layout and appearance -->
  - \usepackage{chngcntr}
  - \counterwithout{table}{chapter}
  - \counterwithout{figure}{chapter}
  - \usepackage[margins = raggedright]{floatrow}
  - \renewcommand{\arraystretch}{1.5}
  - \definecolor{lightgray}{gray}{0.95}
  - \let\OldLongtable\longtable
  - \let\OldEndLongtable\endlongtable
  - \renewenvironment{longtable}{\singlespace\footnotesize\OldLongtable}{\OldEndLongtable}
  - \KOMAoptions{captions=nooneline}
  - \usepackage[
      font = normalsize,
      labelfont = {normalsize,sc},
      justification = justified,
      format = plain
    ]{caption}
  - \usepackage[section]{placeins}
---
```

A file to test can be found here: https://gist.github.com/salim-b/de89b88730e411192476c2875de207e7

**Comment by mangecoeur** [comment]
ouch, you have some seriously complicated yaml! This is most likely because i use some simple regex to identify the yaml heading, so this probably confuses it :/ Thanks for the test text I will try to fix this.

**Comment by salim-b** [comment]
> ouch, you have some seriously complicated yaml!

Told ya my .md files will catch all the edge cases 😜

**Comment by mangecoeur** [comment]
Actually, it's not the yaml it's just i forgot to adjust the line numbers for the jump-to function after stripping out the yaml. Will try to fix this soon.

**Comment by mangecoeur** [comment]
Should be fixed in 1.6.17
